nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-actions.js

```diff
@@ -1086,183 +1497,173 @@
         },
 
         /**
-         * Deletes the flow files in the specified connection.
+         * Deletes the flow files inside the selected connections.
          *
          * @param {type} selection
          */
-        emptyQueue: function (selection) {
-            if (selection.size() !== 1 || !nfCanvasUtils.isConnection(selection)) {
-                return;
-            }
+        emptySelectedQueues: function (selection) {
+            var connections = selection.filter(function (d) {
+                var selectionItem = d3.select(this);
+                return nfCanvasUtils.isConnection(selectionItem);
+            });
+
+            var actionName = selection.size() > 1 ? 'Empty Selected Queues' : 'Empty Selected Queue';
+
+            var dialogContent = selection.size() > 1
+                ? 'Are you sure you want to empty the selected queues? All FlowFiles waiting at the time of the request will be removed.'
+                : 'Are you sure you want to empty the selected queue? All FlowFiles waiting at the time of the request will be removed.';
 
             // prompt the user before emptying the queue
             nfDialog.showYesNoDialog({
-                headerText: 'Empty Queue',
-                dialogContent: 'Are you sure you want to empty this queue? All FlowFiles waiting at the time of the request will be removed.',
+                headerText: actionName,
+                dialogContent: dialogContent,
```

Comment:
Are you sure about this?

Reply:
Yep! Now the `headerText` and `dialogContent` are initialized depending on the specific "empty queue" menu item the user clicked on